### PR TITLE
[release] Move xgboost tune small + microbenchmark release test to new release automation

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/xgboost_tests/xgboost_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    slack: "@tune-team"
+
+  cluster:
+    app_config: app_config.yaml
+    compute_template: tpl_cpu_small.yaml
+
+  run:
+    timeout: 600
+    script: workloads/tune_small.py
+    results: /tmp/tune_small.json
```
Comment (ericl) [inline release/xgboost_tests/xgboost_tests.yaml:13]: Should this be passed in as an env var for the script, so it isn't hard-coded in two separate places?

Comment (krfricke) [reply release/xgboost_tests/xgboost_tests.yaml:13]: For artifacts we have the same problem - it's up to the test creator to make sure these names are synced. I'd keep it that way for now, as the `results` key can also be empty here (then no results will be fetched and just pass/no pass is reported)

Comment (krfricke) [reply release/xgboost_tests/xgboost_tests.yaml:13]: Ok, if the `results` key is set, that value will be used (and set as the TEST_OUTPUT_JSON env variable), and if it is not set, a random `/tmp/output.json` file will be used (and also passed as the TEST_OUTPUT_JSON)

**Comment by krfricke** [comment]
This test now also adds a microbenchmark test. It works locally, but it requires updates to the microbenchmark cli script.

To test this on buildkite, we should merge this PR and supply the link to the wheels afterwards.

**Review by ericl — APPROVED** [review]

**Pull request merged**